src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+        # create the filename where the model will be saved.
+        task_hash_params = dict(
+            benchmark=self.benchmark,
+            task_id=self.task_id,
+            seed=self.seed,
+            **asdict(self.train_config),
+        )
+        logger.info(f"Task hash params: {task_hash_params}")
+        task_hash = compute_identity(**task_hash_params)
+
+        filename = f"{task_hash}.pkl"
+
+        self.checkpoint_file = self.checkpoint_dir / filename
+        logger.info(f"Checkpoint file for this task: {self.checkpoint_file}")
+
+        self.device: torch.device
+        if isinstance(device, torch.device):
+            self.device = device
+        else:
+            self.device = torch.device(
+                device or ("cuda" if torch.cuda.is_available() else "cpu")
+            )
+
+        # TODO: Save those in the configuration dict?
+        self._np_rng: Optional[np.random.RandomState] = None
+        self._torch_rng_state: Optional[Tensor] = None
+        with self.seed_randomness():
+            if os.path.exists(self.checkpoint_file):
+                logger.info(
+                    f"Model has already been trained: loading it from file {self.checkpoint_file}."
+                )
+                self.net, self.h = load_task_network(self.checkpoint_file)
+            else:
+                logger.info(
+                    f"Checkpoint file {self.checkpoint_file} doesn't exist: re-training the model."
+                )
+                logger.info(f"Task hash params: {task_hash_params}")
+
+                self.checkpoint_file.parent.mkdir(exist_ok=True, parents=True)
+                # Need to re-train the meta-model and sample this task.
+                self.net, self.h = get_task_network(
+                    self.input_dir,
+                    self.benchmark,
+                    seed=self.seed,
+                    task_id=self.task_id,
+                    config=self.train_config,
+                )
+                save_task_network(
+                    self.checkpoint_file, self.benchmark, self.net, self.h
+                )
+
+        self.net = self.net.to(device=self.device)
+        self.net.eval()
+        self.h_tensor = torch.as_tensor(self.h, dtype=torch.float32, device=self.device)
+
+        self._space: Space = SpaceBuilder().build(self.get_search_space())
+        # IDEA: Could 'hack' the `sample` function of this Space instance so it always uses our
+        # numpy rng state.
+        # _original_sample = self._space.sample
+
+        # def seeded_sample(_, n_samples: int = 1, seed=None):
+        #     nonlocal self
+        #     if seed is None:
+        #         seed = self._np_rng
+        #     return _original_sample(n_samples=n_samples, seed=seed)
+
+        # self._space.sample = seeded_sample
+
+        self.name = f"profet.{type(self).__qualname__.lower()}_{self.task_id}"
+
+    @contextmanager
+    def seed_randomness(self):
```

Comment:
This could be a standalone helper function. We should either expose seed arguments for each RNG we seed so that they can be seeded independently, or you the provided global seed to sample integers that are used as seeds for the different RNGs to seed. Otherwise we may have weird artifacts as they are synched with the same seed.

Reply:
Each RNG should not be seeded with the same seed. We should use the seed instead to generate random integers that will be used to seed them. Or an alternative, is to expose the seeds in the interface. So instead of passing `seed`, the user would pass `np_seed` and `torch_seed`.

Reply:
Fixed: See the context manager: Using a single seed, and seeding numpy/pytorch based on samples from that first seed.

Reply:
All good, thanks!